File: backend/src/main/java/com/devsu/bank/service/CuentaService.java
package com.devsu.bank.service;

import com.devsu.bank.dto.CuentaRequest;
import com.devsu.bank.dto.CuentaResponse;

import java.util.List;

public interface CuentaService {
    CuentaResponse crear(CuentaRequest request);
    List<CuentaResponse> listarTodos();
    CuentaResponse buscarPorId(Long id);
    CuentaResponse actualizar(Long id, CuentaRequest request);
    CuentaResponse actualizarParcial(Long id, CuentaRequest request);
    void eliminar(Long id);
}

File: frontend/src/app/services/cuenta.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Cuenta, CuentaRequest } from '../models/cuenta.model';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })
export class CuentaService {
  private readonly http = inject(HttpClient);
  private readonly url = `${environment.apiUrl}/cuentas`;

  listar(): Observable<Cuenta[]> {
    return this.http.get<Cuenta[]>(this.url);
  }

  buscarPorId(id: number): Observable<Cuenta> {
    return this.http.get<Cuenta>(`${this.url}/${id}`);
  }

  crear(request: CuentaRequest): Observable<Cuenta> {
    return this.http.post<Cuenta>(this.url, request);
  }

  actualizar(id: number, request: CuentaRequest): Observable<Cuenta> {
    return this.http.put<Cuenta>(`${this.url}/${id}`, request);
  }

  actualizarParcial(id: number, request: Partial<CuentaRequest>): Observable<Cuenta> {
    return this.http.patch<Cuenta>(`${this.url}/${id}`, request);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.url}/${id}`);
  }
}
